habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -191,15 +191,20 @@ def _get_observation_space(self, *args: Any, **kwargs: Any):
         return spaces.Box(
             low=np.iinfo(np.uint32).min,
             high=np.iinfo(np.uint32).max,
-            shape=(self.config.HEIGHT, self.config.WIDTH),
-            dtype=np.uint32,
+            shape=(self.config.HEIGHT, self.config.WIDTH, 1),
+            dtype=np.int32,
         )
 
     def get_observation(
         self, sim_obs: Dict[str, Union[np.ndarray, bool, "Tensor"]]
     ) -> VisualObservation:
         obs = cast(Optional[VisualObservation], sim_obs.get(self.uuid, None))
         check_sim_obs(obs, self)
+        # make semantic observation a 3D array
+        if isinstance(obs, np.ndarray):
+            obs = obs[..., None].astype(np.int32)
```

Comment:
Why do we convert the type to int32 here? Confused why it's necessary.

Reply:
@Skylion007 if I don't convert it here then I'll have to do it in the trainer. Torch doesn't support uint32 so when you initialize the rollout storage with semantic observations that have type np.uint32 it breaks. I moved the conversion here to avoid doing it every time in trainer.

Reply:
Interesting, is this a bug in our habitat-sim code? WE should be returning the same dtype for numpy as torch if possible?

Reply:
@Skylion007 I think so. In my experiments I had to manually typecast to np.int32 to use semantic input for the policy